csc.exe exits with code -2146232797 when attempting to concatenate strings that are part of an inline array

**Version Used**: 
```
.NET SDK:
 Version:           8.0.100
 Commit:            57efcf1350
```
```
<PropertyGroup>
  <OutputType>Exe</OutputType>
  <TargetFramework>net8.0</TargetFramework>
  <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
  <Platforms>AnyCPU;x64</Platforms>
  <LangVersion>12</LangVersion>
</PropertyGroup>
```

`Visual Studio 2022 v17.8.0`

**Steps to Reproduce**:

```
using System;
using System.Runtime.CompilerServices;

class Program {
    public static void Main(string[] args) {
        Test();
    }

    static void Test() {
        var buffer = new ThreeStringBuffer();
        Console.WriteLine(buffer[0] + " " + buffer[1] + " " + buffer[2]);
    }
}

[InlineArray(3)]
struct ThreeStringBuffer {
    string _;
}
```

**Diagnostic Id**:

`MSB6006`

**Expected Behavior**:

The program compiles without issue.

**Actual Behavior**:

"C:\Program Files\Microsoft Visual Studio\2022\Community\MSBuild\Current\Bin\Roslyn\Microsoft.CSharp.Core.targets(84,5): error MSB6006: "csc.exe" exited with code -2146232797."

Diagnostic build output attached: [msb6006.txt](https://github.com/dotnet/roslyn/files/13420007/msb6006.txt)

--- a/src/Compilers/CSharp/Test/Emit2/Semantics/InlineArrayTests.cs
+++ b/src/Compilers/CSharp/Test/Emit2/Semantics/InlineArrayTests.cs
@@ -21903,5 +21903,36 @@ struct MyArray
 }
 ");
         }
+
+        [ConditionalFact(typeof(CoreClrOnly))]
+        [WorkItem("https://github.com/dotnet/roslyn/issues/70910")]
+        public void StringConcatenation()
+        {
+            var src = @"
+using System;
+using System.Runtime.CompilerServices;
+
+class Program
+{
+    public static void Main(string[] args)
+    {
+        Test();
+    }
+
+    static void Test()
+    {
+        var buffer = new ThreeStringBuffer();
+        Console.WriteLine(buffer[0] + ""123"" + buffer[1] + ""124 "" + buffer[2]);
+    }
+}
+
+[InlineArray(3)]
+struct ThreeStringBuffer {
+    string _;
+}
+";
+            var comp = CreateCompilation(src, targetFramework: TargetFramework.Net80, options: TestOptions.ReleaseExe);
+            var verifier = CompileAndVerify(comp, expectedOutput: "123124").VerifyDiagnostics();
+        }
     }
 }

